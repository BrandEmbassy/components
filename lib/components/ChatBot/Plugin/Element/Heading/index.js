'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SUBTITLE_HEADING_LEVEL = exports.TITLE_HEADING_LEVEL = undefined;
exports.default = Heading;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./index.css');

var _index2 = _interopRequireDefault(_index);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TITLE_HEADING_LEVEL = exports.TITLE_HEADING_LEVEL = 1;
var SUBTITLE_HEADING_LEVEL = exports.SUBTITLE_HEADING_LEVEL = 2;

function Heading(_ref) {
  var content = _ref.content,
      _ref$level = _ref.level,
      level = _ref$level === undefined ? TITLE_HEADING_LEVEL : _ref$level;

  if (level === TITLE_HEADING_LEVEL) {
    return _react2.default.createElement(
      'h1',
      { className: _index2.default.Heading },
      content
    );
  }
  return _react2.default.createElement(
    'h2',
    { className: _index2.default.SubHeading },
    content
  );
}