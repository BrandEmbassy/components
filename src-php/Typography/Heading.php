<?php declare(strict_types = 1);

namespace BrandEmbassy\Components\Typography;

use BrandEmbassy\Components\ArrayRenderer;
use BrandEmbassy\Components\EnumValuesToString;
use BrandEmbassy\Components\UiComponent;
use BrandEmbassy\Components\Utilities\UtilitiesOption;
use function is_array;

class Heading implements UiComponent
{
    /**
     * @var UiComponent[]|string[]
     */
    private $children;

    /**
     * @var string
     */
    private $utilityOptions;

    /**
     * @var string
     */
    private $level;


    /**
     * @param UiComponent[]|string[]|UiComponent|string $children
     * @param UtilitiesOption[]                         $utilityOptions
     */
    public function __construct($children, array $utilityOptions, HeadingLevel $level)
    {
        $this->children = is_array($children) ? $children : [$children];
        $this->utilityOptions = EnumValuesToString::transform($utilityOptions);
        $this->level = $level->getValue();
    }


    public function render(): string
    {
        return '<h' . $this->level . ' class="' . $this->utilityOptions . '">'
            . ArrayRenderer::render($this->children)
            . '</h' . $this->level . '>';
    }
}
