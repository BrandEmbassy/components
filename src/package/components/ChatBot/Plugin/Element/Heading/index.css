.Heading {
  font-size : 16px;
  padding: 2px 10px;
  margin: 0;
  display: block;
}
.SubHeading {
  font-size : 13px;
  padding: 2px 10px;
  margin: 0;
  display: block;
}
