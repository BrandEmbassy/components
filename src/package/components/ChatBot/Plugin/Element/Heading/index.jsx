import React from 'react'
import styles from './index.css'

export const TITLE_HEADING_LEVEL = 1
export const SUBTITLE_HEADING_LEVEL = 2

export type HeadingElementProps = {
  level: TITLE_HEADING_LEVEL | SUBTITLE_HEADING_LEVEL,
  content: string
}

export default function Heading ({ content, level = TITLE_HEADING_LEVEL }: HeadingElementProps) {
  if (level === TITLE_HEADING_LEVEL) {
    return <h1 className={styles.Heading}>{content}</h1>
  }
  return <h2 className={styles.SubHeading}>{content}</h2>
}
