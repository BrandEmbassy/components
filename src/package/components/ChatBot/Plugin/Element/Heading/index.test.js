import React from 'react';
import Heading from './index'

describe('Heading', () => {
  const levels = [1, 2, 3]

  levels.forEach((level) =>Â {
    it(`should return heading with correct level ${level}`, () => {
      const wrapper = shallow(<Heading level={level} />)
      epxect(wrapper).toMatchSnapshot()
    });
  });

});
