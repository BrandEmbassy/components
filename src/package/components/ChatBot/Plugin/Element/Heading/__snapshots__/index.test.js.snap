// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should return heading with correct level 1 1`] = `<h1 />`;

exports[`Heading should return heading with correct level 2 1`] = `<h2 />`;

exports[`Heading should return heading with correct level 3 1`] = `<h2 />`;
